### PR TITLE
Critical values

### LINKED ISSUE #16: Implement critical values
Find a place or create a place to implement critical high and low values

It has to be in a place where it is accessible from the model/model manager

also add methods in model/model manager so it is possible to change the critical values (high/low)

In model manager when a new temperature reading is take from a thermometer, and addtemperature method is called. If the new temperature exceeds any of the critical temperatures, call property.firePropertyChange() with "critical" high or low and from what id 
could be:
property.firePropertyChange("Warning")
send the temperature that was measured
what is the limit that was exceeded

### PR DESCRIPTION
**Pull request by hanchry**

### CODE REVIEW

File: src/model/Limits.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+public class Limits {
+  private double upper;
+  private double lower;
+
+  public Limits(double upper, double lower){
+    this.upper = upper;
+    this.lower = lower;
+  }
+
+  public void setLower(double lower) {
```
Comment (chooseanother) [inline src/model/Limits.java]: make it throw maybe exception so that we can put that exception into an errorlabel

File: src/model/Limits.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  public double getUpper() {
+    return upper;
+  }
+
+  public boolean isOverUpperLimit(double value){
+    return value > upper;
+  }
+
+  public boolean isOverLowerLimit(double value){
```
Comment (chooseanother) [inline src/model/Limits.java]: isOverLowerLimit -> isUnderLowerLimit

**Review by chooseanother — CHANGES_REQUESTED** [review]

File: src/model/Limits.java (hunk 2 above)
Comment (chooseanother) [reply src/model/Limits.java]: or below

Comment (hanchry) [reply src/model/Limits.java]: method is changed

File: src/model/Limits.java (hunk 1 above)
Comment (hanchry) [reply src/model/Limits.java]: exeptions are added

File: src/model/Limits.java (hunk 2 above)
Comment (chooseanother) [reply src/model/Limits.java]: isBelowLowerLimit

Comment (hanchry) [reply src/model/Limits.java]: its changed to isUnderLowerLimit

File: src/model/Limits.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+public class Limits {
+  private double upper;
+  private double lower;
+
+  public Limits(double upper, double lower){
+    this.upper = upper;
+    this.lower = lower;
+  }
+
+  public void setLower(double lower) throws Exception {
```
Comment (chooseanother) [inline src/model/Limits.java:12]: These should be accessible from the model/modlemanager

File: src/model/Limits.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      if (lower < upper)
+        this.lower = lower;
+      else{
+        throw new Exception("value to set is to high");
+      }
+  }
+
+  public void setUpper(double upper) throws Exception {
```
Comment (chooseanother) [inline src/model/Limits.java:20]: These should be accessible from the model/modlemanager

File: src/mediator/ModelManager.java [hunk 5]
```diff
@@ -12,11 +13,13 @@
     private Heater heater;
     private PropertyChangeSupport property;
     private TemperatureList temperatureList;
+    private Limits limits;
 
     public ModelManager()
     {
         temperatureList = new TemperatureList();
         property = new PropertyChangeSupport(this);
+        limits = new Limits(0,50);
```
Comment (chooseanother) [inline src/mediator/ModelManager.java]: Maybe a more realistic limit
what is critical temperatures in your room?

**Review by chooseanother — CHANGES_REQUESTED** [review]

File: src/model/Limits.java (hunk 3 above)
Comment (hanchry) [reply src/model/Limits.java:12]: added

File: src/mediator/ModelManager.java (hunk 5 above)
Comment (hanchry) [reply src/mediator/ModelManager.java:22]: added

File: src/model/Limits.java (hunk 4 above)
Comment (hanchry) [reply src/model/Limits.java:20]: added

**Review by chooseanother — APPROVED** [review]

**Pull request merged**